cmd/api/src/database/saved_queries_permissions.go

```diff
@@ -52,16 +69,77 @@ func (s *BloodhoundDB) CreateSavedQueryPermissionToPublic(ctx context.Context, q
-	result := s.db.WithContext(ctx).Where("query_id = ? AND user_id = ?", queryID, userID).Or("query_id = ? AND public = true", queryID).Limit(1)
+	rows := int64(0)
+	result := s.db.WithContext(ctx).Table("saved_queries_permissions").Select("*").Where("query_id = ? AND shared_to_user_id = ?", queryID, userID).Or("query_id = ? AND public = true", queryID).Limit(1).Count(&rows)
 
-	return result.RowsAffected > 0, CheckError(result)
+	return rows > 0, CheckError(result)
 }
 
 // GetPermissionsForSavedQuery gets all permissions associated with the provided query ID
-func (s *BloodhoundDB) GetPermissionsForSavedQuery(ctx context.Context, queryID int64) (model.SavedQueriesPermissions, error) {
-	queryPermissions := model.SavedQueriesPermissions{QueryID: queryID}
+func (s *BloodhoundDB) GetPermissionsForSavedQuery(ctx context.Context, queryID int64) ([]model.SavedQueriesPermissions, error) {
+	queryPermissions := make([]model.SavedQueriesPermissions, 0)
 	result := s.db.WithContext(ctx).Where("query_id = ?", queryID).Find(&queryPermissions)
+
 	return queryPermissions, CheckError(result)
 }
+
+// GetScopeForSavedQuery will return a map of the possible scopes given a query id and a user id
+func (s *BloodhoundDB) GetScopeForSavedQuery(ctx context.Context, queryID int64, userID uuid.UUID) (SavedQueryScopeMap, error) {
+	scopes := SavedQueryScopeMap{
+		SavedQueryScopePublic: false,
+		SavedQueryScopeOwned:  false,
+		SavedQueryScopeShared: false,
+	}
+
+	// Check if the query was shared with the user publicly
+	publicCount := int64(0)
+	if result := s.db.WithContext(ctx).Select("*").Table("saved_queries_permissions").Where("public = true AND query_id = ?", queryID).Count(&publicCount).Limit(1); result.Error != nil {
+		return scopes, result.Error
+	} else if publicCount > 0 {
+		scopes[SavedQueryScopePublic] = true
+	}
+
+	// Check if the user owns the query
+	ownedCount := int64(0)
+	if result := s.db.WithContext(ctx).Select("*").Table("saved_queries").Where("id = ? AND user_id = ?", queryID, userID).Count(&ownedCount).Limit(1); result.Error != nil {
+		return scopes, result.Error
+	} else if ownedCount > 0 {
+		scopes[SavedQueryScopeOwned] = true
+	}
+
+	// Check if the user has had the query shared to them
+	sharedCount := int64(0)
+	if result := s.db.WithContext(ctx).Select("*").Table("saved_queries_permissions").Where("query_id = ? AND shared_to_user_id = ?", queryID, userID).Count(&sharedCount).Limit(1); result.Error != nil {
+		return scopes, result.Error
+	} else if sharedCount > 0 {
+		scopes[SavedQueryScopeShared] = true
+	}
+
+	return scopes, nil
+}
+
+// DeleteSavedQueryPermission deletes the saved query permission associated with the passed in permission id
+func (s *BloodhoundDB) DeleteSavedQueryPermission(ctx context.Context, permissionID int64) error {
+	return CheckError(s.db.WithContext(ctx).Table("saved_queries_permissions").Delete(&model.SavedQueriesPermissions{}, permissionID))
```

Comment:
Is this missing a where clause? If we hit this with an empty id, does it wipe the entire table? Might be beneficial to toss a Limit(1) on this just in case.